egs/dirha_wsj/asr1/run.sh

```diff
@@ -0,0 +1,302 @@
+#!/bin/bash
+
+# Copyright 2017 Johns Hopkins University (Shinji Watanabe)
+# Copyright 2019 Johns Hopkins University (Ruizhi Li)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+. ./path.sh || exit 1;
+. ./cmd.sh || exit 1;
+
+# general configuration
+backend=pytorch
+stage=0        # start from 0 if you need to start from data preparation
+stop_stage=100
+ngpu=1         # number of gpus ("0" uses cpu, otherwise use gpu)
+debugmode=1
+dumpdir=dump   # directory to dump full features
+N=0            # number of minibatches to be used (mainly for debugging). "0" uses all minibatches.
+verbose=0      # verbose option
+resume=        # Resume the training from snapshot
+seed=1
+
+# feature configuration
+do_delta=false
+
+# config files
+preprocess_config=conf/no_preprocess.yaml  # use conf/specaug.yaml for data augmentation
+train_config=conf/train.yaml
+lm_config=conf/lm.yaml
+decode_config=conf/decode.yaml
+
+# rnnlm related
+use_wordlm=true     # false means to train/use a character LM
+lm_vocabsize=65000  # effective only for word LMs
+lm_resume=          # specify a snapshot file to resume LM training
+lmtag=              # tag for managing LMs
+
+# decoding parameter
+n_average=10 # use 1 for RNN models
+recog_model=model.acc.best # set a model to be used for decoding: 'model.acc.best' or 'model.loss.best'
+
+# data
+mic="Beam_Circular_Array"  # Beam_Circular_Array Beam_Linear_Array KA6 L1C
+dirha_wsj_folder=/export/b08/ruizhili/data/Data_processed
+wsj1=/export/corpora5/LDC/LDC94S13B
+
+# exp tag
+tag="" # tag for managing experiments.
+
+. utils/parse_options.sh || exit 1;
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+train_set=train_si284_$mic
+train_dev=dirha_sim_$mic
+train_test=dirha_real_$mic
+recog_set=dirha_real_$mic
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    ### Task dependent. You have to make data the following preparation part by yourself.
+    ### But you can utilize Kaldi recipes in most cases
+    echo "stage 0: Data preparation"
+
+    # If YOU DON'T HAVE AUGMENTED DATA, FOLLOW THE FOLLOWING TWO STEPS BEFORE YOU RUN THE SCRIPT
```

Comment:
I think it's better to prepare the data generation script for other people than CLSP.

Reply:
Sure, I will work on that. Thanks.

Reply:
prepared the data generation script